lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GaussianProcessConditionalCovariance.cxx

```diff
@@ -0,0 +1,550 @@
+{
+
+  const UnsignedInteger inputDimension = xi.getDimension();
+  const CovarianceModel covarianceModel(result_.getCovarianceModel());
+  if (inputDimension != covarianceModel.getInputDimension())
+    throw InvalidArgumentException(HERE) << " In GaussianProcessConditionalCovariance::getConditionalMarginalVariance, input data should have the same dimension as covariance model's input dimension. Here, (input dimension = " << inputDimension << ", covariance model spatial's dimension = " << covarianceModel.getInputDimension() << ")";
+  if (!indices.check(covarianceModel.getOutputDimension()))
+    throw InvalidArgumentException(HERE) << "In GaussianProcessConditionalCovariance::getConditionalMarginalVariance, the indices of a marginal sample must be in the range [0," << covarianceModel.getOutputDimension()
+                                         << " ] and must be different";
+  const UnsignedInteger sampleSize = xi.getSize();
+  if (sampleSize == 0)
+    throw InvalidArgumentException(HERE) << " In GaussianProcessConditionalCovariance::getConditionalMarginalVariance, expected a non empty sample";
+  // Compute the matrix & return only the marginalIndex diagonal element
+  Sample result(inputDimension, indices.getSize());
+  Point data(inputDimension);
+  for (UnsignedInteger i = 0; i < sampleSize; ++i)
+  {
+    for (UnsignedInteger j = 0; j < inputDimension; ++j) data[j] = xi(i, j);
+    result[i] = getConditionalMarginalVariance(data, indices);
+  }
+  return result;
+}
+
+
+/* Compute joint normal distribution conditionally to observations*/
+Normal GaussianProcessConditionalCovariance::operator()(const Point & xi) const
+{
+  Sample sample(1, xi);
+  return operator()(sample);
+}
+
+/* Compute joint normal distribution conditionally to observations*/
+Normal GaussianProcessConditionalCovariance::operator()(const Sample & xi) const
+{
+  // The Normal distribution is defined by its mean & covariance
+  LOGINFO("In GaussianProcessConditionalCovariance::operator() : evaluating the mean");
+  const Sample meanAsSample = getConditionalMean(xi);
+  // Mean should be a Point ==> data are copied form the Sample to a Point
+  const Point mean(meanAsSample.getImplementation()->getData());
+  LOGINFO("In GaussianProcessConditionalCovariance::operator() : evaluating the covariance");
+  const CovarianceMatrix covarianceMatrix = getConditionalCovariance(xi);
+  // Check the covariance matrix. Indeed, if point is very similar to one of the learning points, covariance is null
+  // Even if this check is done in Normal::Normal, we perform debugging
+  LOGINFO("In GaussianProcessConditionalCovariance::operator() : evaluating the Normal distribution");
+  if (!covarianceMatrix.isPositiveDefinite()) throw InvalidArgumentException(HERE) << "In GaussianProcessConditionalCovariance::operator(), the covariance matrix is not positive definite. The given points could be very close to the learning set. Could not build the Normal distribution";
```

Comment:
We already have a regularization process and a test in EllipticalDistribution.cxx to deal with the singular case. No need to have this test here.